src/db/tables/onerep_scans.ts

```diff
@@ -76,16 +76,27 @@ async function addOnerepScanResults(
         .delete()
         .where("onerep_scan_id", onerepScanId);
     } else {
-      // Initial and Monitoring scans always create a new scan:
-      await transaction("onerep_scans").insert({
-        onerep_profile_id: onerepProfileId,
-        onerep_scan_id: onerepScanId,
-        onerep_scan_reason: onerepScanReason,
-        onerep_scan_status: onerepScanStatus,
-        // @ts-ignore knex.fn.now() results in it being set to a date,
-        // even if it's not typed as a JS date object:
-        created_at: knex.fn.now(),
-      });
+      // Initial and Monitoring scans create a new scan if they don’t
+      // exist already. Existing scans get their status updated.
+      await transaction("onerep_scans")
+        .insert({
+          onerep_profile_id: onerepProfileId,
+          onerep_scan_id: onerepScanId,
+          onerep_scan_reason: onerepScanReason,
+          onerep_scan_status: onerepScanStatus,
+          // @ts-ignore knex.fn.now() results in it being set to a date,
+          // even if it's not typed as a JS date object:
+          created_at: knex.fn.now(),
+        })
+        .onConflict("onerep_scan_id")
+        .ignore()
+        .where("onerep_scan_id", onerepScanId)
+        .update({
+          onerep_scan_status: onerepScanStatus,
+          // @ts-ignore knex.fn.now() results in it being set to a date,
+          // even if it's not typed as a JS date object:
+          updated_at: knex.fn.now(),
+        });
```

Comment:
Woah, does this work? Does this result in a single query? I'd have expected something [like `.onConflict().merge()`](https://knexjs.org/guide/query-builder.html#onconflict):

```suggestion
        .merge(["updated_at", "onerep_scan_status"]);
```

Reply:
Had to update the columns explicitly with the new values — otherwise the linter complained about non-matching types: https://github.com/mozilla/blurts-server/pull/3470/commits/6c597731e2daa5d914bfcfbbf0ad38659a251a75.

Reply:
That works for me!